From c7c7b44971b06da29d7e7a570ff304ac3aa4b686 Mon Sep 17 00:00:00 2001
From: "Christensen, Taylor" <taylor.christensen@intel.com>
Date: Mon, 9 Nov 2015 16:22:35 -0800
Subject: [PATCH 075/441] SDHCI: Enable HW Mutex in acpi mode

Enable the HW mutex in SDHCI ACPI mode. This should, in
theory lock and unlock the mutex during ACPI SDHCI calls.
It is in an ifdef else statement so if the flag is not set,
it will work as normal without it.
---
 drivers/mmc/host/sdhci-acpi.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

--- a/drivers/mmc/host/sdhci-acpi.c
+++ b/drivers/mmc/host/sdhci-acpi.c
@@ -47,6 +47,10 @@
 
 #include "sdhci.h"
 
+#ifdef CONFIG_HW_MUTEXES
+#include <linux/hw_mutex.h>
+#endif
+
 enum {
 	SDHCI_ACPI_SD_CD	= BIT(0),
 	SDHCI_ACPI_RUNTIME_PM	= BIT(1),
@@ -427,8 +431,21 @@ static int sdhci_acpi_probe(struct platf
 	}
 
 	host->mmc->caps2 |= MMC_CAP2_NO_PRESCAN_POWERUP;
+	
+#ifdef CONFIG_HW_MUTEXES
+
+	LOCK_EMMC_HW_MUTEX(host->mmc);
 
 	err = sdhci_add_host(host);
+
+	UNLOCK_EMMC_HW_MUTEX(host->mmc);
+
+#else 
+
+	err = sdhci_add_host(host);
+
+#endif
+	
 	if (err)
 		goto remove_slot;
 
